app/screens/home/search/initial/initial.tsx

```diff
@@ -22,9 +24,11 @@ type Props = {
     teamId: string;
     teamName: string;
     teams: TeamModel[];
+    crossTeamSearchEnabled: boolean;
 }
 
-const Initial = ({recentSearches, scrollEnabled, searchValue, setRecentValue, searchRef, teamId, teamName, teams, setTeamId, setSearchValue}: Props) => {
+const Initial = ({recentSearches, scrollEnabled, searchValue, setRecentValue, searchRef, teamId, teamName, teams, setTeamId, setSearchValue, crossTeamSearchEnabled}: Props) => {
+    const showRecentSearches = Boolean(recentSearches.length) && teamId !== ALL_TEAMS_ID;
```

Comment:
does this mean that we won't be showing any searches that were previously performed on all teams? why?

Reply:
The current system to store the recent searches requires the team (referenced by its `ID`) to exist in the DB, to make this work we would need to change how they are stored (remove the requirement for a team to exist in the DB) or make a dedicated table for `all teams` recent searches.
I made a ticket in the Epic ([MM-62341](https://mattermost.atlassian.net/browse/MM-62341)) so we can prioritize and not forget about it. There's also some discussions about having the equivalent feature on the web side that might change how we want to handle this.